src/sybil/query_spec.go

```diff
@@ -103,13 +105,13 @@ func (qs *QuerySpec) NewResult() *Result {
 	return addedRecord
 }
 
-func (master_result *ResultMap) Combine(results *ResultMap) {
+func (master_result *ResultMap) Combine(results *ResultMap, fullMerge bool) {
 	for k, v := range *results {
 		mval, ok := (*master_result)[k]
 		if !ok {
 			(*master_result)[k] = v
 		} else {
-			mval.Combine(v)
+			mval.Combine(v, fullMerge)
```

Comment:
why fullMerge passed in like this. maybe another way to do this, like setting an option on master_result